static/css/mozsearch.css

```diff
@@ -119,37 +119,48 @@ td.code {
 td#line-numbers {
     padding: 0;
 }
+
+.source-line-with-number {
+  padding: 0rem 1rem;
+  line-height: 1.3;
+  display: flex;
+}
+
+/* In order to display the line numbers and have them visible to screen readers
+   but not appear in any copy-and-pasted selection, we create a before
+   pseudo-element with generated content. */
+.line-number::before {
+    content: attr(data-line-number);
+}
 .line-number {
-    display: block;
+    display: inline-block;
     cursor: pointer;
-    -moz-user-select: none;
-    -o-user-select: none;
-    -khtml-user-select: none;
-    -webkit-user-select: none;
-    -ms-user-select: none;
     user-select: none;
     text-align: right;
     padding: 0 0.5rem;
-    position: relative;
+    /* this was originally set on the ".file td:first-child" */
+    color: #aaa;
+    width: 8ex;
 }
 .highlighted,
 .multihighlight {
     background: none repeat scroll 0 0 rgb(255, 255, 204) !important;
 }
-td.code {
-    display: block;
+.source-line {
     margin: 0;
     padding: 0;
-    position: relative;
-    z-index: 1;
+    display: inline-block;
+    white-space: pre;
+    flex-grow: 1;
 }
-td pre code {
-    padding: 0 0 0 1em;
-    margin: 0;
-    padding: 0;
-    display: block;
+.source-line-with-number.stuck {
+  background: #ddf;
+}
+.source-line-with-number.last-stuck {
+  background: linear-gradient(#ddf, #ddf 90%, blue);
```

Comment:
In the case where a line is both selected and the last-stuck, the blue line (which I believe comes from this background) seems to be hidden by the yellow of the selection. Is there a way around that? I played around with the interaction between selection and sticky and other than this minor issue it behaves perfectly. Nice work!

Reply:
Yeah, if we add a selector like:
```css
.last-stuck .multihighlight {
	background: linear-gradient(rgb(255, 255, 204), rgb(255, 255, 204) 90%, blue) !important;
}
```

Then we get the yellow with a blue line at the bottom of the .last-stuck thing.  I also need to do something similar for the blame blocks which I think look a little weird without the blue line.  My plan for now is to introduce CSS variables for the highlight color, the sticky background color, and the last-stuck line at the bottom.

I'll also add comments to the CSS explaining the use of the gradient as a way to display a border line without affecting the layout box of the source lines, with it also being something we can potentially cross-animate without any layout impact.

Reply:
Totally drive-by, but outline or inset-but-solid box shadows are usually a better fit for that (depending on whether you want the fake border outside or inline the box).

Reply:
Great suggestion, thank you!  I'll convert to that.

Reply:
Actually, outline can also be inside the box with negative outline offset, like: `data:text/html,<div style="width: 100px; height: 100px; outline: 10px solid red; outline-offset: -10px"></div>`, and it is probably more efficient and easier to reason about.

Reply:
@emilio Hm, so if I use:
```css
.source-line-with-number.last-stuck,
.last-stuck .highlighted,
.last-stuck .multihighlight,
.last-stuck div .blame-strip {
  box-shadow: inset 0 -1px 0 0 var(--sticky-header-bottom-border-color);
}
```

Then I get the eternal anti-aliased line drawing problem that sometimes the line is 1 device pixel thick and sometimes it's 2 pixels thick.  This was with my "layout.css.devPixelsPerPx" at 1.25 and after resetting it back to "-1.0" (and squinting a little :)

Reply:
I had originally considered outline but was concerned that it's on all sides and general concern about accessibility/focus rings.  I did just give it a quick try and was surprised to realize that using a normal 1px outline somewhat does the right thing simply because the top side gets clipped at the top of the first .stuck line and on all the sides, and then the background colors of the neighboring .stuck elements hide things.

I'm not quite sure what you're proposing with your example where the outline is 10px wide though.

Reply:
Sorry for all the churn here.  I hadn't restarted after resetting my devPixelsPerPx and it seems that I properly get a single pixel line from an inset box-shadow once I do.  However, I either need to go back to 1.25 or I need to change my default font or zoom size or something.

Reply:
I've reverted to using the gradient approach.  It works with apparent visual consistency across all of my machines I tested on:
- Linux with gnome-shell under X11 with fonts scaling factor of 1.25
  - Nightly with layout.css.devPixelsPerPx=1.25
  - Nightly with layout.css.devPixelsPerPx=-1.0
- OS X 15" 2018 Macbook with default settings:
  - Nightly with layout.css.devPixelsPerPx=-1.0
- Windows 10 with 100% scaling:
  - Release with layout.css.devPixelsPerPx=-1.0

I did try using a value like -1.6px instead of 1.6px since that would equate to the 90% gradient transition point I chose, but that still resulted in oscillating line widths.

In any event, I do think we can do better than the gradient hack, I just think we should maybe defer that to a follow-up.  With any luck someone with better design sensibility and/or better understanding of Firefox graphics rendering and how devPixelsPerPx will step in :wink: